### ISSUE TITLE
UnicodeDecodeError parsing ClinVar VCF file

### ISSUE THREAD

**Issue by freekvh** [jamescasbon/PyVCF#268]
Hi,

I downloaded the file clinvar_20170130.vcf.gz from ftp://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh38/

I start extracting information but I hit a UnicodeDecodeError:

```
file = vcf.Reader(filename='clinvar_20170130GRCh38.vcf.gz')
genelist = []
for record in file:
    genelist.append(Record.INFO['GENEINFO'].split(':')[0])
```

```
UnicodeDecodeError                        Traceback (most recent call last)
<ipython-input-42-a2a406474c55> in <module>()
      1 genelist = []
----> 2 for record in file:
      3     genelist.append(Record.INFO['GENEINFO'].split(':')[0])

/cadappl/python3/anaconda-3.5.1/lib/python3.5/site-packages/vcf/parser.py in __next__(self)
    540     def __next__(self):
    541         '''Return the next record in the file.'''
--> 542         line = next(self.reader)
    543         row = self._row_pattern.split(line.rstrip())
    544         chrom = row[0]

/cadappl/python3/anaconda-3.5.1/lib/python3.5/site-packages/vcf/parser.py in <genexpr>(.0)
    278         self._alt_pattern = re.compile('[\[\]]')
    279 
--> 280         self.reader = (line.strip() for line in self._reader if line.strip())
    281 
    282         #: metadata fields from header (string or hash, depending)

/cadappl/python3/anaconda-3.5.1/lib/python3.5/codecs.py in __next__(self)
    640 
    641         """ Return the next decoded line from the input stream."""
--> 642         line = self.readline()
    643         if line:
    644             return line

/cadappl/python3/anaconda-3.5.1/lib/python3.5/codecs.py in readline(self, size, keepends)
    553         # If size is given, we call read() only once
    554         while True:
--> 555             data = self.read(readsize, firstline=True)
    556             if data:
    557                 # If we're at a "\r" read one extra character (which might

/cadappl/python3/anaconda-3.5.1/lib/python3.5/codecs.py in read(self, size, chars, firstline)
    499                 break
    500             try:
--> 501                 newchars, decodedbytes = self.decode(data, self.errors)
    502             except UnicodeDecodeError as exc:
    503                 if firstline:

UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 52: ordinal not in range(128)
```

At the moment the parsing crashes the genelist is 10937 entries long so it does work for a lot of entries. also:
```

Record = next(file)
print('Chrom =', Record.CHROM)
print('POS =', Record.POS)
print('ID =', Record.ID)
print('REF =', Record.REF)
print('ALT =', Record.ALT)
print('Qual =', Record.QUAL)
print('Filter =', Record.FILTER)
print('Info =', Record.INFO)
print('\n')
print('format =', Record.FORMAT)
print('Samples =', Record.samples)
print('Record =', Record.genotype)
#print('\n')
print('Info =', Record.INFO['GENEINFO'])
```

Works and gives me meaningful data. Is one entry corrupt?

I tried excepting the error but that doesn't work, it still stops parsing. I have no idea how I can invesigate said single entry.

Any suggestions? Thank you!
(As you can see I'm using Python 3.5.1 and I'm using the latest version of pyvcf.)

**Comment by freekvh** [comment]
Hi. After locally installing the latest Anaconda, I have no more errors, so the error is somewhere else, sorry (now to find out where exactly...)

**Comment by freekvh** [comment]
So I found that the error is not in the version of any software but in the fact that Python (3) on Windows makes difference encoding assumptions than on Linux. I had already given up om PyVCF but I encountered the same error just doing a plain "with open". I found that adding the encoding (like this: with open(input_file, 'r', encoding="utf-8")) fixes the error, it probably also fixes the error from PyVCF. At the moment I can't check if the encoding parameter is supported by the read function in PyVCF and fixes things but I suspect so.

In the mean time a colleague confirmed that most characters in my VCF file are indeed ascii but some are UTF8 (most/all are letters with accents on them like  è, é, ê, ë).
